examples/vault-prod-ready/main.tf

```diff
@@ -0,0 +1,534 @@
+  to_port                  = 8500
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.elb_consul.id
+  security_group_id        = module.consul_cluster.security_group_id
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY THE VAULT SERVER CLUSTER
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "vault_cluster" {
+  # When using these modules in your own templates, you will need to use a Git URL with a ref attribute that pins you
+  # to a specific version of the modules, such as the following example:
+  source = "github.com/hashicorp/terraform-aws-vault.git//modules/vault-cluster?ref=v0.13.3"
+
+  cluster_name  = var.vault_cluster_name
+  cluster_size  = var.vault_cluster_size
+  instance_type = var.vault_instance_type
+
+  ami_id    = var.ami_id
+  user_data = data.template_file.user_data_vault_cluster.rendered
+
+  vpc_id     = var.vpc_id
+#  subnet_ids = var.vpc_subnet_ids
+
+  # This setting will create the AWS policy that allows the vault cluster to
+  # access KMS and use this key for encryption and decryption
+  enable_auto_unseal = true
+
+  auto_unseal_kms_key_arn = data.aws_kms_alias.vault-example.target_key_arn
+
+  # To make testing easier, we allow requests from any IP address here but in a production deployment, we *strongly*
+  # recommend you limit this to the IP address ranges of known, trusted servers inside your VPC.
+
+  allowed_ssh_cidr_blocks              = []
+  allowed_inbound_cidr_blocks          = []
+  allowed_inbound_security_group_ids   = []
+  allowed_inbound_security_group_count = 0
+  ssh_key_name                         = var.ssh_key_name
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# ATTACH IAM POLICIES FOR CONSUL
+# To allow our Vault servers to automatically discover the Consul servers, we need to give them the IAM permissions from
+# the Consul AWS Module's consul-iam-policies module.
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "consul_iam_policies_servers" {
+  source = "github.com/hashicorp/terraform-aws-consul.git//modules/consul-iam-policies?ref=v0.7.0"
+
+  iam_role_id = module.vault_cluster.iam_role_id
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# GIVE SSH-GRUNT PERMISSIONS TO TALK TO IAM
+# We add an IAM policy to Jenkins that allows ssh-grunt to make API calls to IAM to fetch IAM user and group data.
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "ssh_grunt_policies" {
+  source = "git::git@github.com:gruntwork-io/module-security.git//modules/iam-policies?ref=v0.19.1"
```

Comment:
`module-security` is a private Gruntwork repo. It should not be used in this public example.